parser-typechecker/src/Unison/Server/CodebaseServer.hs

```diff
@@ -152,23 +152,18 @@ start
   :: Var v => Codebase IO v Ann -> (Strict.ByteString -> Port -> IO ()) -> IO ()
 start codebase k = do
   envToken <- lookupEnv "UCM_TOKEN"
-  envHost <- lookupEnv "UCM_HOST"
-  envPort <- (readMaybe =<<) <$> lookupEnv "UCM_PORT"
-  token <- case envToken of
+  envHost  <- lookupEnv "UCM_HOST"
+  envPort  <- (readMaybe =<<) <$> lookupEnv "UCM_PORT"
+  token    <- case envToken of
     Just t -> return $ C8.pack t
-    _ -> genToken
-
-  let settings = case envHost of
-        Just p -> setHost (fromString p) defaultSettings
-        _ -> defaultSettings
+    _      -> genToken
+  let settings = appEndo
+        (  foldMap (Endo . setPort)              envPort
+        <> foldMap (Endo . setHost . fromString) envHost
+        )
+        defaultSettings
       a = app codebase token
-
-  case envPort of
-    Just p -> do
-      (k token p)
```

Comment:
This starts up UCM and waits for it to finish (which it appears to do when it receives a CTRL-D for some reason.